src/test/java/com/fittracker/fittracker/service/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,90 @@
+package com.fittracker.fittracker.service;
+
+import com.fittracker.fittracker.entity.User;
+import com.fittracker.fittracker.exception.UserAlreadyExistsException;
+import com.fittracker.fittracker.repository.UserRepository;
+import com.fittracker.fittracker.request.RegisterRequest;
+import com.fittracker.fittracker.response.RegisterResponse;
+import com.fittracker.fittracker.security.JwtUtils;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.testcontainers.shaded.com.trilead.ssh2.auth.AuthenticationManager;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AuthenticationServiceTest {
+
+    private static final RegisterRequest registerRequest = new RegisterRequest("user", "user@example.com", "password");
+
+    @Captor
+    private ArgumentCaptor<User> userCaptor;
+
+    @Mock
+    PasswordEncoder passwordEncoder;
+    @Mock
+    UserRepository userRepository;
+    @Mock
+    JwtUtils jwtUtils;
+    @Mock
+    AuthenticationManager authenticationManager;
+
+    @InjectMocks
+    AuthenticationService authenticationService;
+
+    @Nested
+    class Register {
+        @Test
+        void givenNonExistingUser_shouldReturnRegisterResponse() {
+            UUID uuid = UUID.randomUUID();
+
+            when(userRepository.existsByUsernameOrEmail(any(), any())).thenReturn(false);
```

Comment:
let's setup all mocks in `@BeforeEach` with `lenient().when()` to make tests more readable

Reply:
These mocks are not identical:
`when(userRepository.existsByUsernameOrEmail(any(), any())).thenReturn(false);`
`when(userRepository.existsByUsernameOrEmail(any(), any())).thenReturn(true);`

one returns false, other returns true. Is it possible then to put it in `@BeforeEach`?